### ISSUE TITLE
Pool "hangs"

### ISSUE THREAD

**Issue by tgirardi** [andris9/nodemailer-smtp-pool#7]
I have an app. that's been working OK for months, but on the last 3 weeks I noticed that it was reaching a point where it just stopped sending emails. When restarting the app. mails started flowing again, but just for some hours. Eventually that same state was reached and no email was sent ever again.

While debugging I  discovered that `maxConnections` was being reached while having all connections unavailable, so at that point `_processMessages` was not able to create a new connection and returned without sending the email (`return` at line 120).

Even more, connections seem to remain unavailable forever, so the process never exits this state and no mail is sent ever again until restarting the process (each call to `send` ends up with `_processMessages` returning through line 120, without sending the email).

Our pool is created with the following configuration:

````javascript
smtp = nodemailer.createTransport(nodemailesSmtpPool({
            host: ini.smtp.host
          , port: ini.smtp.port
          , auth: {
              user: ini.smtp.user
            , pass: ini.smtp.pass
            }
          , maxConnections: 20
          , maxMessages: 20
          , socketTimeout: 5*60*1000
          , connectionTimeout: 5*60*1000
          }));
`````

We are using nodemailer 1.3.0 and nodemailer-smtp-pool 0.1.8

**Comment by andris9** [comment]
Thanks, it sounds fishy and I'll look into it

**Comment by andris9** [comment]
Similar issue was resolved with 3e7a431e9f2e015c01f1ca0b01230c5a972377c7 (v1.1.2)
